lib/core/connector_job.rb

```diff
@@ -62,16 +63,36 @@ def canceling?
       status == Connectors::SyncStatus::CANCELING
     end
 
+    def suspended?
+      status == Connectors::SyncStatus::SUSPENDED
+    end
+
+    def canceled?
+      status == Connectors::SyncStatus::CANCELED
+    end
+
+    def pending?
+      Connectors::SyncStatus::PENDING_STATUSES.include?(status)
+    end
+
+    def active?
+      Connectors::SyncStatus::ACTIVE_STATUSES.include?(status)
+    end
+
+    def terminated?
+      Connectors::SyncStatus::TERMINAL_STATUSES.include?(status)
+    end
+
     def connector_snapshot
-      self[:connector]
+      self[:connector] || {}
```

Comment:
Why default to an empty object instead of nil?

Reply:
Resolve this conversation, and reply in another comment.